indexer/src/main/java/au/org/aodn/esindexer/utils/MapperUtils.java

```diff
@@ -410,4 +422,90 @@ public static Optional<Contacts> mapContactInfo(List<CIContactPropertyType2> con
+                    }
+                    // Email
+                    if (inv.isPresent() && !inv.get().getEmails().isEmpty()) {
+                        invContactsModel.setEmails(inv.get().getEmails());
+                    } else {
+                        invContactsModel.setEmails(i != null ? i.getEmails() : null);
+                    }
+                    // Phone
+                    if (inv.isPresent() && !inv.get().getPhones().isEmpty()) {
+                        invContactsModel.setPhones(inv.get().getPhones());
+                    } else {
+                        invContactsModel.setPhones(i != null ? i.getPhones() : null);
+                    }
+                    // Online Resources
+                    // Phone
+                    if (inv.isPresent() && !inv.get().getOnlineResources().isEmpty()) {
+                        invContactsModel.setLinks(inv.get().getOnlineResources());
+                    } else {
+                        invContactsModel.setLinks(i != null ? i.getOnlineResources() : null);
+                    }
+
+                    return invContactsModel;
+                })
+                .toList());
+    }
+
+    public static List<ContactsModel> mapOrgContacts(CIResponsibilityType2 ciResponsibility, AbstractCIPartyPropertyType2 party) {
+        List<ContactsModel> results = new ArrayList<>();
+        if(party.getAbstractCIParty() != null
+                && party.getAbstractCIParty().getValue() != null
+                && party.getAbstractCIParty().getValue() instanceof CIOrganisationType2 organisation) {
+            Optional<MapperUtils.Contacts> org = MapperUtils.mapContactInfo(organisation.getContactInfo());
+
+            if(organisation.getIndividual() != null && !organisation.getIndividual().isEmpty()) {
+                results.addAll(mapContactsFromOrg(ciResponsibility, organisation));
+            } else {
+                ContactsModel orgContactsModel = ContactsModel.builder().build();
+                orgContactsModel.setRoles(MapperUtils.mapContactsRole(ciResponsibility));
+                orgContactsModel.setOrganization(MapperUtils.mapContactsOrganization(party));
+                orgContactsModel.setOrganization(organisation.getName().getCharacterString().getValue().toString());
+
+                if(org.isPresent() && !org.get().getAddresses().isEmpty()) {
+                    orgContactsModel.setAddresses(org.get().getAddresses());
+                }
+
+                if(org.isPresent() && !org.get().getEmails().isEmpty()) {
+                    orgContactsModel.setEmails(org.get().getEmails());
+                }
+
+                if(org.isPresent() && !org.get().getPhones().isEmpty()) {
+                    orgContactsModel.setPhones(org.get().getPhones());
+                }
+
+                if(org.isPresent() && !org.get().getOnlineResources().isEmpty()) {
+                    orgContactsModel.setLinks(org.get().getOnlineResources());
+                }
```

Comment:
same, here, overusing `.isPresent()`, consolidate them. 

and apply null check before `.isEmpty()` for each of the field.